.github/workflows/test.yml

```diff
@@ -0,0 +1,73 @@
+name: Test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
```

Comment:
Is there a way to re-use these between files (same with the first few steps below)?

Reply:
I guess it's time to setup a CI Docker image :-)

Reply:
How do we feel about making a new issue for that docker image? Outside of my wheel house and seems significant enough to deserve its own review.

Reply:
Good with me :)

Reply:
yep, can happen later

Reply:
Created #82 for it